src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageActionDescriptorProvider.cs

```diff
@@ -103,27 +112,53 @@ private void AddActionDescriptors(IList<ActionDescriptor> actions, RazorProjectI
 
             foreach (var selector in model.Selectors)
             {
-                actions.Add(new PageActionDescriptor()
+                var order = selector.AttributeRouteModel.Order ?? 0;
+                var attributeRouteInfo = new AttributeRouteInfo()
                 {
-                    AttributeRouteInfo = new AttributeRouteInfo()
+                    Name = selector.AttributeRouteModel.Name,
+                    Order = order,
+                    Template = selector.AttributeRouteModel.Template,
+                };
+                actions.Add(CreateDescriptor(item, model, filters, attributeRouteInfo, item.PathWithoutExtension));
+
+                if (!string.Equals(model.Name, viewEnginePath, StringComparison.Ordinal))
+                {
+                    // If a page has an alias, register an ActionDescriptor with a higher-ordered AttributeRoute.
+                    // The ordering would ensure that the route gets used for link generation, but is never used for inbound routing.
```

Comment:
Just to double check, this scenario is covered by some of the new functional tests, right?